### PR TITLE
fix #293203 + preference rework  (PR #5922 for 4.0): improved responsiveness, ease of adding preferences, UI

### PR DESCRIPTION
**Pull request by SKefalidis**
Resolves: #5922 for 4.0

If this gets merged before #5623 look at #5922 where I've transferred those changes to this new preference style.

This was a big rebase and the line-count is not the same (to be expected thanks to the new style) so I would advise you to review this before merging, I might have missed something (even though I tested it).

- [x] I signed [CLA](https://musescore.org/en/cla)
- [x] I made sure the code in the PR follows [the coding rules](https://github.com/musescore/Documentation/blob/master/WorkflowAndGuidelines/CodeGuidelines.md)
- [x] I made sure the code compiles on my machine
- [x] I made sure there are no unnecessary changes in the code
- [x] I made sure the title of the PR reflects the core meaning of the issue you are solving
- [x] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"
- [x] I made sure the commit message title starts with "fix #424242:" if there is a related issue
- [n\a] I created the test (mtest, vtest, script test) to verify the changes I made

### CODE REVIEW

**Comment by SKefalidis** [comment]
Removed some comments, fixed the formatting of one line and removed a file that shouldn't be here.

**Comment by SKefalidis** [comment]
Added 2 translatable strings (Apply/Applying).

File: mscore/prefsdialog.cpp [hunk 1]
```diff
@@ -1007,130 +1302,95 @@ void PreferenceDialog::apply()
...
+
+    if (buttonBox->button(QDialogButtonBox::Apply)->isEnabled() == false) {
+        return;
     }
+
+    bool uiStyleThemeChanged = false;
+    bool audioModified = false;
+
+    applySetActive(false);
+    buttonBox->button(QDialogButtonBox::Apply)->setText(tr("Applying..."));
```
Comment (Jojo-Schmitz) [inline mscore/prefsdialog.cpp:1316]: Please change to `Applying…`, i.e. with an ellipsis. See also #6201

**Comment by shoogle** [comment]
There's no need for the commit message to include the PR number for the 3.x branch. In fact, it is better if the commit message is the same on both branches.

If you want to include the PR number, its best to put it in the [body of the commit message](https://commit.style/) rather than in the title. Putting it in the title just makes it harder to see what the commit actually does. (Ideally we would also put `fix #xxxx` in the body but it's a bit late to change that now.)

**Comment by Jojo-Schmitz** [comment]
rebase needed

File: mscore/prefsdialog.cpp (hunk 1 above)
Comment (SKefalidis) [reply mscore/prefsdialog.cpp]: Done.

**Comment by SKefalidis** [comment]
Rebased.

**Comment by SKefalidis** [comment]
Well, uncrustify made this way uglier.

**Pull request merged**